#414: Added examples and explanations for features in documentation (#415)
exasol/notebook-connector: doc/user_guide/text-ai-extension-examples.rst

Text AI Extension Examples

> ###########################

Reviewer: ```suggestion
Text AI Extension Examples
##########################
```

Text AI extraction is incremental: it processes only source rows for which no
results have been written yet.  The output tables therefore act as persistent
storage for prior extraction results and are re-used across runs.

Depending on the extractor, TXAIE can create more than one table.  Besides a
main output table, some workflows also create support and lookup tables that
store normalized intermediate results.  This is the same table layout used in
the bundled preprocessing notebooks.

Calling ``extraction.run(my_secrets)`` opens a database connection, activates
the Text AI language container for the session, and executes the extraction
SQL against the configured source and output tables.

.. code-block:: python

    from exasol.nb_connector.text_ai_extension_wrapper import Extraction
    from exasol.ai.text.extractors.named_entity_extractor import NamedEntityExtractor

    # Configure the extractor and the output table
    extraction = Extraction(
        extractor=NamedEntityExtractor(),
        output="MY_SCHEMA.EXTRACTION_RESULTS",
    )

    # Execute the extraction — results are written to the output table
    extraction.run(my_secrets)

Pipeline extraction example

> ***************************

Reviewer: ```suggestion
Pipeline Extraction Example
***************************
```

pipeline in ``Extraction``.

.. code-block:: python

    from exasol.nb_connector.text_ai_extension_wrapper import Extraction
    from exasol.ai.text.extraction.abstract_extraction import Defaults, Output
    from exasol.ai.text.extractors.extractor import PipelineExtractor
    from exasol.ai.text.extractors.source_table_extractor import (
        NameSelector,
        SchemaSource,
        SourceTableExtractor,
        TableSource,
    )
    from exasol.ai.text.extractors.standard_extractor import StandardExtractor

    src_extractor = SourceTableExtractor(
        source=TableSource(
            source=SchemaSource("MY_SCHEMA"),
            table_names=NameSelector(["CUSTOMER_SUPPORT_TICKETS"]),
        )
    )
    std_extractor = StandardExtractor()

    extraction = Extraction(
        extractor=PipelineExtractor(steps=[src_extractor, std_extractor]),
        output=Output(db_schema="MY_SCHEMA"),
        defaults=Defaults(),
    )

    extraction.run(my_secrets)

In this pattern, each pipeline step can create its own output and support
tables.  Re-running the pipeline is incremental as long as the previous output
tables are still present.

Branch extraction example

> *************************

Reviewer: ```suggestion
Branch Extraction Example
*************************
```

Text AI Extension Examples
###########################

The Text AI Extension provides ready-to-use UDFs for named-entity recognition,
zero-shot classification, and semantic feature extraction, powered by
large language models running inside the Exasol database.

**Prerequisites in the SCS** – same as for the Transformers Extension: full
DB connection parameters and full BucketFS parameters (see
:doc:`secret-store-examples`).

Step 1 – Deploy a license
**************************

``deploy_license`` uploads a license file to BucketFS and registers it with
the extension.  The Notebook Connector bundles a community license that covers
non-commercial and evaluation use — call the function with no extra arguments
to use it.  For commercial deployments, supply either the path to a license

> file (``license_file``) or the license XML content as a string

Reviewer 2: Not an xml, yaml with signature 

And call it text-ai license file. Because it is not the DB license file

Author: Thanks , understood, 

fixed wording and examples to refer to a Text AI license file and show YAML style signature content as you said.

(``license_content``).

.. code-block:: python

    from exasol.nb_connector.text_ai_extension_wrapper import deploy_license

    # Use the built-in community license (no arguments required)
    deploy_license(my_secrets)

    # Supply your own license file from disk
    from pathlib import Path
    deploy_license(my_secrets, license_file=Path("license.xml"))

    # Supply the license content directly as a string
    deploy_license(my_secrets, license_content="<license>...</license>")

> Step 2 – Initialize the extension

Reviewer 2: Advice the llm. That we use AP Stylebook titles

Author: updated the headings and titles accordingly

***********************************

``initialize_text_ai_extension`` performs the full installation in one call:

> 1. Downloads the extension's Script Language Container (SLC) from GitHub and

Reviewer 2: Not GitHub, but s3

Author: fixed wordings and kept the description focused on the actual container input and deployment flow.

   uploads it to BucketFS (``install_slc=True``).
2. Downloads the three default Hugging Face models and uploads them to BucketFS
   so that UDFs can use them offline (``install_models=True``).
3. Creates the UDF scripts in the configured schema (``install_scripts=True``).

Any of the three steps can be skipped by passing its flag as ``False``.  This
is useful for incremental updates — e.g., pass ``install_slc=False`` to only
refresh the UDF scripts without re-uploading the container.

You can also pin a specific extension version with ``version=`` or supply a

> locally built container archive with ``container_file=``.

Reviewer 2: ```suggestion
locally downloaded/available container archive with ``container_file=``.
```

Author: Applied

.. code-block:: python

    from exasol.nb_connector.text_ai_extension_wrapper import initialize_text_ai_extension

    # Full installation with latest version
    initialize_text_ai_extension(my_secrets)

    # Pin a specific version
    initialize_text_ai_extension(my_secrets, version="1.2.3")

    # Install from a local container file and skip model download
    from pathlib import Path
    initialize_text_ai_extension(
        my_secrets,
        container_file=Path("/tmp/txaie.tar.gz"),
        install_models=False,
    )

Step 3 – Run an extraction
***************************

> The ``Extraction`` class wraps a UDF call.  Provide an *extractor* that

Reviewer 2: ```suggestion
The ``Extraction`` class wraps multiple UDF calls.  Provide an *extractor* that
```

Author: Applied

> defines which UDF to invoke (e.g. ``NamedEntityExtractor`` for NER) and an

Reviewer 2: ```suggestion
defines which UDFs to invoke (e.g. ``NamedEntityExtractor`` for NER) and an
```

Author: Applied

``output`` table name where results will be written.  Calling
``extraction.run(my_secrets)`` opens a database connection and executes the
extraction SQL, reading from any source table you configure on the extractor.

.. code-block:: python

    from exasol.nb_connector.text_ai_extension_wrapper import Extraction
    from exasol.ai.text.extractors.named_entity_extractor import NamedEntityExtractor

    # Configure the extractor and the output table
    extraction = Extraction(

>         extractor=NamedEntityExtractor(),

Reviewer 2: DefaultExtractor missing and Branch and Pipeline as well, the latter we probably can ignore for the moment 

Also we need to state that text-ai does incremental text extraction, means it runs the extraction for rows from the source for which it hasn't the results already computed before and stored in the database.

Furthermore, we probably need to explain the how the tables work, see the notebooks

Author: I added DefaultExtractor pipeline related contents, incremental extraction behavior, and output table. I did not add a dedicated Branch example or a deeper explanation of the table layout yet.

I will do tat ASAP

Author: fixed.
